lib/apiservers/engine/backends/backends.go

```diff
@@ -359,7 +363,48 @@ func EventService() *events.Events {
 // RegistryCheck checkes the given url against the registry whitelist, blacklist, and insecure
 // registries lists. It returns true for each list where u matches that list.
 func (d *dynConfig) RegistryCheck(ctx context.Context, u *url.URL) (wl bool, bl bool, insecure bool) {
-	m := d.update(ctx)
+	var m *dynConfig
```

Comment:
Please replace this with singleflight as used here:
https://github.com/vmware/vic/blob/master/lib/portlayer/attach/communication/connector.go#L114

I think the current code is racy (particularly the draining of `d.updateCh` and the associated `case` statement - possible fall through with no one actually doing the work)

Reply:
Replaced with singleflight. Also moved this call to `update` since `SystemInfo` was also calling `update`.